tests/testthat/_snaps/dataframe-s3-base.md

```diff
@@ -969,3 +969,126 @@
       │ 3   │
       └─────┘
 
+# Column subsetting with `[[` fails with value = NULL
+
+    Code
+      dat[[value]]
+    Condition
+      Error in `dat[[value]]`:
+      ! Can't subset a polars DataFrame with `[[`.
+      i Subscript `value` must be a string or a non-0 scalar integer, not `NULL`.
+
+# Column subsetting with `[[` fails with value = 1.5
+
+    Code
+      dat[[value]]
+    Condition
+      Error in `dat[[value]]`:
+      ! Can't subset a polars DataFrame with `[[`.
+      i Subscript `value` must be a string or a non-0 scalar integer, not the number 1.5.
+
+# Column subsetting with `[[` fails with value = 0
+
+    Code
+      dat[[value]]
+    Condition
+      Error in `dat[[value]]`:
+      ! Can't subset a polars DataFrame with `[[`.
+      i Subscript `value` must be a string or a non-0 scalar integer, not the number 0.
+
+# Column subsetting with `[[` fails with value = 4
+
+    Code
+      dat[[value]]
+    Condition
+      Error in `x$select()`:
+      ! Evaluation failed in `$select()`.
+      Caused by error:
+      ! Evaluation failed in `$collect()`.
+      Caused by error:
+      ! Column(s) not found: nth
+
+# Column subsetting with `[[` fails with value = -4
+
+    Code
+      dat[[value]]
+    Condition
+      Error in `x$select()`:
+      ! Evaluation failed in `$select()`.
+      Caused by error:
+      ! Evaluation failed in `$collect()`.
+      Caused by error:
+      ! Column(s) not found: nth
```

Comment:
That's a weird error message, can we improve it? (same with value = 4)

Reply:
This uses the error from the `nth` as is, so it would probably be better to improve the nth error message upstream.